bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/BallerinaValues.java

```diff
@@ -76,17 +79,51 @@
      * @param fieldValues values to be used for fields when creating the object value instance.
      * @return value of the object.
      */
+    @Deprecated
     public static ObjectValue createObjectValue(BPackage packageId, String objectTypeName, Object... fieldValues) {
+        return createObjectValue(packageId, objectTypeName, getStrand(), fieldValues);
+    }
+
+    private static ObjectValue createObjectValue(BPackage packageId, String objectTypeName, Strand currentStrand,
+                                                Object... fieldValues) {
+        //This method duplicates the createObjectValue with referencing the issue in runtime API getting strand
         ValueCreator valueCreator = ValueCreator.getValueCreator(packageId.toString());
         Object[] fields = new Object[fieldValues.length * 2];
+        Scheduler scheduler = currentStrand != null ? currentStrand.scheduler : null;
+        State prevState = currentStrand != null ? currentStrand.getState() : State.RUNNABLE;
+        boolean prevBlockedOnExtern = currentStrand != null && currentStrand.blockedOnExtern;
+        ObjectValue returnObjectValue;
 
         // Adding boolean values for each arg
         for (int i = 0, j = 0; i < fieldValues.length; i++) {
             fields[j++] = fieldValues[i];
             fields[j++] = true;
         }
-        //passing scheduler, strand and properties as null for the moment, but better to expose them via this method
-        return valueCreator.createObjectValue(objectTypeName, null, null, null, fields);
+        try {
+            //Check for non-blocking call
+            if (currentStrand != null && currentStrand.getState() != State.RUNNABLE) {
```

Comment:
Isn't it enough to check `currentStrand != null` why do we check `currentStrand.getState() != State.RUNNABLE` ?